modules/features/referrals/src/main/kotlin/au/com/shiftyjelly/pocketcasts/referrals/ReferralsSendGuestPassPage.kt

```diff
@@ -44,24 +47,42 @@ import au.com.shiftyjelly.pocketcasts.compose.images.SubscriptionBadge
 import au.com.shiftyjelly.pocketcasts.referrals.ReferralPageDefaults.pageCornerRadius
 import au.com.shiftyjelly.pocketcasts.referrals.ReferralPageDefaults.pageWidthPercent
 import au.com.shiftyjelly.pocketcasts.referrals.ReferralPageDefaults.shouldShowFullScreen
+import au.com.shiftyjelly.pocketcasts.sharing.AnalyticsData
+import au.com.shiftyjelly.pocketcasts.sharing.SharingClient
+import au.com.shiftyjelly.pocketcasts.sharing.SharingRequest
 import au.com.shiftyjelly.pocketcasts.ui.theme.Theme
 import au.com.shiftyjelly.pocketcasts.utils.extensions.getActivity
+import kotlinx.coroutines.launch
 import au.com.shiftyjelly.pocketcasts.images.R as IR
 import au.com.shiftyjelly.pocketcasts.localization.R as LR
 
 @OptIn(ExperimentalMaterial3WindowSizeClassApi::class)
 @Composable
 fun ReferralsSendGuestPassPage(
+    sharingClient: SharingClient,
     onDismiss: () -> Unit,
 ) {
     AppTheme(Theme.ThemeType.DARK) {
         val context = LocalContext.current
         val windowSize = calculateWindowSizeClass(context.getActivity() as Activity)
+        val scope = rememberCoroutineScope()
 
         ReferralsSendGuestPassContent(
             windowWidthSizeClass = windowSize.widthSizeClass,
             windowHeightSizeClass = windowSize.heightSizeClass,
             onDismiss = onDismiss,
+            onShare = {
+                val referralCode = "test_code" // TODO - Referrals: Make it dynamic
+                val request = SharingRequest.referralLink(
+                    referralCode = referralCode,
+                )
+                    .setSourceView(SourceView.REFERRALS)
+                    .setAnalyticsData(AnalyticsData(AnalyticsEvent.REFERRAL_LINK_SHARED, mapOf("code" to referralCode)))
```

Comment:
Since the `referralCode` is passed to the factory method it would be better to set it there. This way caller doesn't have to worry about it. Something like

```kotlin
Builder(Data.ReferralLink(referralCode))
    .setAnalyticsData(
        AnalyticsData(AnalyticsEvent.REFERRAL_LINK_SHARED, 
        mapOf("code" to referralCode))
    )
```